multinet/api/models/network.py

```diff
@@ -79,6 +82,102 @@ def create_with_edge_definition(
             ],
         )
 
+        # default behavior is to run Pregel algorithms when a new network is created.  This stores 
+        # data back in the node table.  If there is no node table, prohibit running the algorithms
+
+        if run_graph_analyses and (len(node_tables)>0):
+
+            try:
+                # automatically add graph measurements to the graph by running analysis jobs
+                arango_db = workspace.get_arango_db(readonly=False)
+                print("running page rank analysis on:",name)
+                pagerank_job_id = arango_db.pregel.create_job(
+                    graph=name,
+                    algorithm='pagerank',
+                    store=True,
+                    #maxx_gss=100,
+                    #thread_count=1,
+                    async_mode=False,
+                    result_field='_pagerank'
+                )
+                # a conflict writing results were noticed when multiple jobs finish about the same time, so 
+                # wait after each job before starting the next one
+
+                while (arango_db.pregel.job(pagerank_job_id)['state'] == 'running'):
```

Comment:
Looks like there is a state for `storing` the data back to the db

Check here for the possible states: https://www.arangodb.com/docs/stable/http/pregel.html#get-pregel-job-execution-status

You will probably want to update all the instances that we are waiting for a job to finish

Reply:
Thanks for noticing the `storing` state.  I am adding this into the code now and will report during our call.  Correct, that I just haven't pushed the version with the bound variables yet.  Thanks for testing this.

Reply:
I just pushed a commit to add bound variables and add a test for `storing` data after the job is finished.  I'll look into whether  arango supports a write lock to allow the jobs to run in parallel and then write sequentially.   This passes all the demo datasets.  

In our meeting today, it was suggested to move this code into a task executed after a network is generated.

Reply:
TODO:  I see why we need to separate this code into a separate task.  It also needs to be executed in the upload task in csv.py when I network is created by the UI.